pam/integration-tests/vhs-helpers_test.go

```diff
@@ -49,7 +49,9 @@ var (
 		"AUTHD_SLEEP_EXAMPLE_BROKER_QRCODE_WAIT": 4 * time.Second,
 	}
 
-	vhsSleepRegex = regexp.MustCompile(`(?m)\$\{?(AUTHD_SLEEP_[A-Z_]+)\}?(\s?([*/]+)\s?([\d.]+))?.*$`)
+	vhsSleepRegex = regexp.MustCompile(
+		`(?m)\$\{?(AUTHD_SLEEP_[A-Z_]+)\}?(\s?([*/]+)\s?([\d.]+))?.*$`)
+	vhsEmptyLinesRegex = regexp.MustCompile(`(?m)((^\n^\n)+(^\n)?|^\n)(^─+$)`)
```

Comment:
There are also some duplicate empty lines which are not above a separator, for example in `pam/integration-tests/testdata/TestCLIChangeAuthTok/golden/prevent_change_password_if_user_does_not_exist`. We could also remove those:


```diff
diff --git a/pam/integration-tests/vhs-helpers_test.go b/pam/integration-tests/vhs-helpers_test.go
index ffdce45b..5ad1f923 100644
--- a/pam/integration-tests/vhs-helpers_test.go
+++ b/pam/integration-tests/vhs-helpers_test.go
@@ -51,7 +51,8 @@ var (
 
        vhsSleepRegex = regexp.MustCompile(
                `(?m)\$\{?(AUTHD_SLEEP_[A-Z_]+)\}?(\s?([*/]+)\s?([\d.]+))?.*$`)
-       vhsEmptyLinesRegex = regexp.MustCompile(`(?m)((^\n^\n)+(^\n)?|^\n)(^─+$)`)
+       vhsEmptyLinesBeforeSeparatorRegex = regexp.MustCompile(`(?m)((^\n^\n)+(^\n)?|^\n)(^─+$)`)
+       vhsDuplicateEmptyLinesRegex       = regexp.MustCompile(`(?m)(\n){3,}`)
 )
 
 func newTapeData(tapeName string, settings ...tapeSetting) tapeData {
@@ -192,7 +193,9 @@ func (td tapeData) ExpectedOutput(t *testing.T, outputDir string) string {
        got = permissionstestutils.IdempotentPermissionError(got)
 
        // Drop all the empty lines before each page separator, to remove the clutter.
-       got = vhsEmptyLinesRegex.ReplaceAllString(got, "$4")
+       got = vhsEmptyLinesBeforeSeparatorRegex.ReplaceAllString(got, "$4")
+       // Drop all duplicate empty lines
+       got = vhsDuplicateEmptyLinesRegex.ReplaceAllString(got, "\n\n")
 
        // Save the sanitized result on cleanup
        t.Cleanup(func() {
```

Reply:
I need to check but I'd prefer not to remove duplicate lines that are in between UI elements or any other stuff happening before the program being finished as it may hide issues in the program being launched or in our output.

So not doing this was on purpose.

Reply:
So, I've checked the golden you mention, but that's exactly one of those cases we should fix in the bubbletea rendering, not in the golden files.

Reply:
> I'd prefer not to remove duplicate lines that are in between UI elements or any other stuff happening before the program being finished as it may hide issues in the program being launched or in our output.

Yeah, I see that.

> So, I've checked the golden you mention, but that's exactly one of those cases we should fix in the bubbletea rendering, not in the golden files.

Ok, works for me.